### PR TITLE
test: move to tests folder and add admin delete test

### PR DESCRIPTION
**Pull request by rehmanis**
- restructure tests to move them into separate `tests` folder
- add admin delete test
- update Role model and the User-Role association table named `role_user_link`. Previously deleting a user would do a cascade delete i.e remove all roles associated with it even if the a role was also associated with another user which was not deleted. Since we want the role to still exist for other users, when we delete a user, only the user object  and any mappings for that user in association table is deleted from database

Note: make sure run migration and update or delete the `migrations` folder and `shopyo.db` file and re initialize the app

### CODE REVIEW

File: shopyo/conftest.py [hunk 1]
```diff
@@ -130,7 +132,7 @@ def auth(test_client):
     return AuthActions(test_client)
 
 
-class AuthActions(object):
+class AuthActions():
```
Comment (Abdur-rahmaanJ) [inline shopyo/conftest.py]: Remove (), only `class AuthActions:`

File: shopyo/modules/box__default/admin/tests/test_admin.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 """
 from flask import request
 from flask import url_for
-from modules.box__default.admin.models import Role, User
+from modules.box__default.admin.models import Role, User, role_user_link
```
Comment (Abdur-rahmaanJ) [inline shopyo/modules/box__default/admin/tests/test_admin.py:13]: Wrong past practise, can you add each import on new line?

from modules.box__default.admin.models import Role
from modules.box__default.admin.models import User
from modules.box__default.admin.models import role_user_link

File: shopyo/conftest.py (hunk 1 above)
Comment (rehmanis) [reply shopyo/conftest.py]: Done

File: shopyo/modules/box__default/admin/tests/test_admin.py (hunk 2 above)
Comment (rehmanis) [reply shopyo/modules/box__default/admin/tests/test_admin.py:13]: Done. Is there a something like pycodestyle to check this for us?

**Comment by rehmanis** [comment]
Can you hold on from merging? I need to add one more thing.

File: shopyo/modules/box__default/admin/tests/test_admin.py (hunk 2 above)
Comment (Abdur-rahmaanJ) [reply shopyo/modules/box__default/admin/tests/test_admin.py:13]: @rehmanis hummm no ... just a clean style for large code bases.

**Comment by rehmanis** [comment]
> Can you hold on from merging? I need to add one more thing.

Good to go.

**Comment by Abdur-rahmaanJ** [comment]
@rehmanis bump version in __init__.py . small fixes:  _ . _ . x, feature, many fixes etc: _ . x . _, big feature, breaking change etc x . _ . _

**Comment by rehmanis** [comment]
> @rehmanis bump version in **init**.py . small fixes: _ . _ . x, feature, many fixes etc: _ . x . _, big feature, breaking change etc x . _ . _

Sounds Good. You mean for next commit onwards correct? Didn't know about this thanks for mentioning.

**Comment by Abdur-rahmaanJ** [comment]
Yes, me too i forget ^^

**Pull request merged**